src/components/Customer/CartDialog.vue

```diff
@@ -212,25 +229,31 @@ export default {
     },
     async populatePoints() {
       const res = await axios.get("/customer/account");
-      if (res.status == 200 || res.status == 304) {
+      if (res.status == 200) {
         this.points = res.data[0].points;
       }
     },
     async populateRecentAddresses() {
       const res = await axios.get("/customer/frequents");
-      if (res.status == 200 || res.status == 304) {
+      if (res.status == 200) {
         this.recentAddresses = res.data;
       }
     },
     async populatePromos() {
       const res = await axios.get(`/customer/promos/${this.rid}`);
-      if (res.status == 200 || res.status == 304) {
+      if (res.status == 200) {
+        this.promos.push("None");
```

Comment:
Shouldn't this option still be there if fetching of promos fail?

Reply:
It should not populate if it fails.